File: pizza/files/default/generate-consul-template.rb
#!/usr/bin/env ruby
require "json"

CONSUL_JSON = "/etc/consul/consul.json"

if ARGV.size < 2
  puts "usage: ruby gen-consul-template.rb [consul_machine_ip] [consul_server_ip]"
  exit 1
end

consul_content = File.read(CONSUL_JSON)
consul_config = JSON.parse(consul_content)

consul_config["advertise_addr"] = ARGV[0]
consul_config["start_join"][0] = ARGV[1]

pretty_config = JSON.pretty_generate(consul_config)
puts pretty_config

File.open(CONSUL_JSON, "w") { |file| file.write(pretty_config) }

File: pizza/files/default/generate-consul-template.py
#!/usr/bin/env python
import json
import sys

CONSUL_JSON = "/etc/consul/consul.json"

if len(sys.argv) < 2:
    print("usage: ruby gen-consul-template.rb [consul_machine_ip] [consul_server_ip]")
    sys.exit(1)

with open(CONSUL_JSON) as f:
    consul_content = f.read()
    consul_config = json.loads(consul_content)

    consul_config["advertise_addr"] = sys.argv[1]
    consul_config["start_join"][0] = sys.argv[2]

    pretty_config = json.dumps(consul_config, indent=4, sort_keys=True)
    print(pretty_config)

with open(CONSUL_JSON, "w") as f:
    f.write(pretty_config)
